src/main/python/systemml/defmatrix.py

```diff
@@ -764,33 +1184,31 @@ def _aggFn(self, fnName, axis):
         """
         Common function that is called for functions that have axis as parameter.
         """
-        dmlOp = DMLOp([self])
-        out = matrix(None, op=dmlOp)
+        dml_script = ''
+        lhsStr, inputs = _matricize(self, [])
         if axis is None:
-            dmlOp.dml = [out.ID, ' = ', fnName, '(', self.ID, ')\n']
+            dml_script = [OUTPUT_ID, ' = ', fnName, '(', lhsStr, ')\n']
         else:
-            dmlOp.dml = [out.ID, ' = ', fnName, '(', self.ID, ', axis=', str(axis) ,')\n']
-        return out
+            dml_script = [OUTPUT_ID, ' = ', fnName, '(', lhsStr, ', axis=', str(axis) ,')\n']
+        return construct_intermediate_node(inputs, dml_script)
 
     ######################### Indexing operators ######################################
 
     def __getitem__(self, index):
         """
         Implements evaluation of right indexing operations such as m[1,1], m[0:1,], m[:, 0:1]
         """
-        dmlOp = DMLOp([self])
-        out = matrix(None, op=dmlOp)
-        dmlOp.dml = [out.ID, ' = ', self.ID ] + getIndexingDML(index) + [ '\n' ]
-        return out
+        print '__getitem__'
```

Comment:
Can you update this to Python 3 compatible syntax, as this currently throws an error as soon as the `systemml` package is imported.